packages/react-account/src/context.tsx

```diff
@@ -1,34 +1,30 @@
 import React from "react";
 import { useRefCallback } from "@crossbell/util-hooks";
-
-export type BaseSigner = {
-	signMessage: (msg: string) => Promise<string>;
-	getAddress: () => Promise<string | null>;
-};
+import { type WalletClient } from "wagmi";
 
 export type ReactAccountContext = {
 	onDisconnect: () => void;
-	getSinger: () => Promise<BaseSigner>;
+	getWalletClient: () => Promise<WalletClient>;
```

Comment:
We pass `getSinger` to `react-account` due to the issues `wagmi` encounters with React Native. Consequently, `xlog-mobile` utilizes `@ethersproject/providers` and `@walletconnect/react-native-dapp` for wallet connections and message signing [(Source Code)](https://github.com/Crossbell-Box/xLog-mobile/blob/main/src/providers/connect-kit-provider/index.tsx#L19-L27), rather than relying on `wagmi`.

Is there any reason to switch from `getSinger` to `getWalletClient`?

Reply:
See https://wagmi.sh/react/migration-guide#provider--signer-terminology

In wagmi v1, there's no signer, but `WalletClient` is basically `Signer`

Reply:
The problem is, `@ethersproject/providers` and `@walletconnect/react-native-dapp` are not compatible with `WalletClient`. Previously, we introduced a general type called `BaseSigner` to address type conflicts.

I noticed that `signMessage` from `WalletClient` now has a new parameter type: `{ account: string; message: string }`. A potential solution is to modify the `BaseSigner` type to `signMessage({ account: string; message: string }): Promise<string>` while keeping the rest of the code unchanged.

By doing so, all `xlog-mobile` needs to do is implement a new `signMessage` method that conforms to the updated interface. Otherwise, it would be difficult for `xlog-mobile` to meet `WalletClient` requirements.

Reply:
Done.

Reply:
BTW, `account` in `signMessage` is optional, so it's unnecessary to change the call signature.